fl_training.py

```diff
@@ -375,24 +368,29 @@ def compute_test_score(
     if is_save_fig:
 
         # Save data
-        np.save(save_folder / "score_matrix", score_matrix)
-        np.save(save_folder / "global_val_acc", global_val_acc)
-        np.save(save_folder / "global_val_loss", global_val_loss)
-        np.save(save_folder / "score_matrix_extended", score_matrix_extended)
-
+        metrics_results = {}
+        metrics_results["score_matrix"] = score_matrix
+        metrics_results["global_val_acc"] = global_val_acc
+        metrics_results["global_val_loss"] = global_val_loss
+        metrics_results["score_matrix_extended"] = score_matrix_extended
+        with open(save_folder / "metrics_results.p", 'wb') as f:
```

Comment:
Since we have both metrics and loss, I would name the file something like history_data (it's the name used by Keras)